src/util/std_expr.h

```diff
@@ -4283,47 +4283,109 @@ class infinity_exprt : public nullary_exprt
 };
 
 /// \brief A let expression
-class let_exprt : public ternary_exprt
+class let_exprt : public binary_exprt
 {
 public:
-  let_exprt(symbol_exprt symbol, exprt value, const exprt &where)
-    : ternary_exprt(
+  class let_binding_exprt : public binary_exprt
```

Comment:
Another question: why is a `let_binding_exprt` not at all related to a `binding_exprt`? That seems surprising to me.

Reply:
This is not trivial: there's only one "where", and not one per binding, whereas the let bindings need an additional "value". Thus, the "is-a" relationship is not easy to establish.

Reply:
Ack, but then the `binding_exprt` may need to be reconsidered if it isn't actually as generic as was hoped? Maybe it should just not be called `binding_exprt` and instead its API should be folded into that of a `quantifier_exprt`? I would just like to avoid confusion.

Reply:
I did consider that, but it does fit `array_comprehension_exprt`. That would then need to be a quantifier (odd) or have it's own API.

Reply:
Another idea. An option could be that `let_exprt` gets a `binding_exprt` as operand!

Reply:
That would seem nice and consistent with the other uses of `binding_exprt`!

Reply:
Ok, will go for it!

Reply:
done